vmware/govmomi
# Add vAPI bindings for vLCM - offline depots and cluster baseline images

## Description

This PR adds bindings for parts of the vLCM REST API. More specifically for the methods that allow the manipulation of offline depots and of cluster baseline images.

These bindings are necessary for the implementation of vLCM automation in https://github.com/hashicorp/terraform-provider-vsphere

Links to the doc entries for each method included in this change are provided in the source code.

## Type of change

Please mark options that are relevant:

  not work as expected)
- [ ] Build related change

## How Has This Been Tested?

Added handlers in the simulator and a new bats test
I have also tested the new bindings on a live environment. Below you can find a scenario that goes through every new method in order:

Preparation steps
1. Deploy a vCenter server and an ESXi host
2. Move the host into a cluster
3. Upload a software bundle (drivers, vibs, etc.) to a location where it is accessible via http/https

Test steps. Call the following methods
1. "CreateOfflineDepot" - to create a new depot with your software bundle
2. "GetOfflineDepots" - to see the list of depots. Copy the identifier of your new depot
3. [Optional] "GetOfflineDepot" - to see the details of an individual depot
4. "GetOfflineDepotContent" - to obtain the identifier and version of your component (driver/vib/something)
5. [Optional] "ListSoftwareDrafts" - to see if there are any drafts
6. [Optional] "DeleteSoftwareDraft" - to clean up any unfinished drafts
7. "CreateSoftwareDraft" - to open a new baseline draft
8. "UpdateSoftwareDraftComponents" - to add your components to the draft
9. [Optional] "GetSoftwareDraftComponent" - to check if everything is okay in the draft
10. "CommitSoftwareDraft" - to commit and apply your draft
11. "CreateSoftwareDraft" - to create a second draft
12. "RemoveSoftwareDraftComponents" - to remove your component from the image
13. "CommitSoftwareDraft" - to commit and apply your second draft
14. "DeleteOfflineDepot" - to delete the offline depot from step 1

## Checklist:


[guidelines]: https://github.com/vmware/govmomi/blob/main/CONTRIBUTING.md
